KalturaWowzaServer/src/main/java/com/kaltura/media_server/services/KalturaAPI.java

```diff
@@ -292,6 +296,22 @@ public static KalturaLiveAsset getliveAsset(KalturaFlavorAssetListResponse liveA
         return null;
     }
 
+    public void sendBeacon(String entryId, int partnerId, String data, String beaconTag) throws Exception {
+        if (partnerId == -5) {
+            KalturaClient Client = getClient();
+            KalturaLiveEntry liveEntry = Client.getLiveStreamService().get(entryId);
+            partnerId = liveEntry.partnerId;
+        }
+
+        KalturaBeacon beacon = new KalturaBeacon();
+        beacon.relatedObjectType = KalturaBeaconObjectTypes.ENTRY_BEACON;
+        beacon.eventType = beaconTag;
+        beacon.objectId = entryId;
+        beacon.privateData = data;
 
+        KalturaClient impersonateClient = impersonate(partnerId);
```

Comment:
will also add try catch on everything, since we don't realy care if it was working or not